components/observations/submitObservation.ts

```diff
@@ -0,0 +1,115 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import axios from 'axios';
+import * as FileSystem from 'expo-file-system';
+import {isUndefined, omit} from 'lodash';
+import md5 from 'md5';
+
+import {ObservationFormData} from 'components/observations/ObservationFormData';
+import {AvalancheCenterID, MediaItem, mediaItemSchema, Observation} from 'types/nationalAvalancheCenter';
+import {apiDateString} from 'utils/date';
+
+const extensionToMimeType = (extension: string) => {
+  switch (extension.toLowerCase()) {
+    case 'jpeg':
+    case 'jpg':
+      return 'image/jpeg';
+    case 'png':
+      return 'image/png';
+    default:
+      throw new Error(`Unknown mime type for extension: ${extension}`);
+  }
+};
+
+const imageUploadCachePrefix = 'IMAGE_UPLOAD_CACHE';
+
+// TODO: should put an expiration time on these cache entries and clear them less aggressively.
+export const clearUploadCache = async () => {
+  const keys = await AsyncStorage.getAllKeys();
+  keys.filter(k => k.startsWith(imageUploadCachePrefix)).map(async k => await AsyncStorage.removeItem(k));
+};
+
+const uploadImage = async ({apiPrefix, uri, name, center_id}: {apiPrefix: string; center_id: AvalancheCenterID; uri: string; name: string | undefined}): Promise<MediaItem> => {
+  // This weird use of `slice` is because the version of Hermes that Expo is currently pinned to doesn't support `at()`
+  const filename = uri.split('/').slice(-1)[0];
```

Comment:
Do windows phones use the other slash? :trollface:

Reply:
URIs can't have backslashes but nice try 😉